java/ingest/ingest-batcher-submitter/src/test/java/sleeper/ingest/batcher/submitter/IngestBatcherSubmitterLambdaIT.java

```diff
@@ -0,0 +1,329 @@
+            String json = "{" +
+                    "\"files\":[\"" + TEST_BUCKET + "/test-directory\"]," +
+                    "\"tableName\":\"" + TEST_TABLE + "\"" +
+                    "}";
+
+            // When
+            lambda.handleMessage(json, RECEIVED_TIME);
+
+            // Then
+            assertThat(store.getAllFilesNewestFirst())
+                    .containsExactly(
+                            fileRequest(TEST_BUCKET + "/test-directory/nested-2/test-file-2.parquet"),
+                            fileRequest(TEST_BUCKET + "/test-directory/nested-1/test-file-1.parquet"));
+        }
+    }
+
+    @Nested
+    @DisplayName("Store files from multiple file paths")
+    class StoreFilesFromMultipleFilePaths {
+        @Test
+        void shouldStoreFilesWhenJsonHasMultipleFilePaths() {
+            // Given
+            uploadFileToS3("test-file-1.parquet");
+            uploadFileToS3("test-file-2.parquet");
+            String json = "{" +
+                    "\"files\":[" +
+                    "\"test-bucket/test-file-1.parquet\"," +
+                    "\"test-bucket/test-file-2.parquet\"" +
+                    "]," +
+                    "\"tableName\":\"test-table\"" +
+                    "}";
+
+            // When
+            lambda.handleMessage(json, RECEIVED_TIME);
+
+            // Then
+            assertThat(store.getAllFilesNewestFirst())
+                    .containsExactly(
+                            fileRequest("test-bucket/test-file-2.parquet"),
+                            fileRequest("test-bucket/test-file-1.parquet"));
+        }
+
+        @Test
+        void shouldSkipFileIfNotFound() {
+            // Given
+            uploadFileToS3("test-file-2.parquet");
+            String json = "{" +
+                    "\"files\":[" +
+                    "\"test-bucket/not-found.parquet\"," +
+                    "\"test-bucket/test-file-2.parquet\"" +
+                    "]," +
+                    "\"tableName\":\"test-table\"" +
+                    "}";
+
+            // When
+            lambda.handleMessage(json, RECEIVED_TIME);
+
+            // Then
+            assertThat(store.getAllFilesNewestFirst())
+                    .containsExactly(
+                            fileRequest("test-bucket/test-file-2.parquet"));
+        }
+    }
+
+    @Nested
+    @DisplayName("Handle errors in messages")
+    class HandleErrorsInMessage {
+
+        @Test
+        void shouldIgnoreAndLogMessageWithInvalidJson() {
+            // Given
+            String json = "{";
+
+            // When
+            lambda.handleMessage(json, RECEIVED_TIME);
+
+            // Then
+            assertThat(store.getAllFilesNewestFirst()).isEmpty();
+        }
+
+        @Test
+        void shouldIgnoreAndLogMessageIfTableDoesNotExist() {
+            // Given
+            String json = "{" +
+                    "\"files\":\"[" + TEST_BUCKET + "/test-file-1.parquet\"]," +
```

Comment:
This JSON is invalid. It's a bit difficult to notice because of how the bucket name is added to the string, but there's a double quote character before the start of the array.

We could update these tests to give the JSON as a full literal rather than adding the bucket and table name like this?